lms/services/canvas_api.py

```diff
@@ -50,6 +51,22 @@ def get_token(self, authorization_code):
             response.parsed_params.get("expires_in"),
         )
 
+    def get_refreshed_token(self, refresh_token):
+        response = self._helper.validated_response(
+            self._helper.refresh_token_request(refresh_token),
+            CanvasRefreshTokenResponseSchema,
+        )
+
+        new_access_token = response.parsed_params["access_token"]
+
+        self._save(
+            new_access_token,
+            response.parsed_params.get("refresh_token", refresh_token),
+            response.parsed_params.get("expires_in"),
+        )
+
+        return new_access_token
```

Comment:
Code that calls this is going to need the new access token right away so return it rather than making the caller have to query the DB for it